client/state/document-head/selectors.js

```diff
@@ -39,38 +34,17 @@ export function getUnreadCount( state ) {
  * @return {?String}        Formatted title
  */
 export function getFormattedTitle( state ) {
-	const unreadCount = getUnreadCount( state );
-	const titleAndSiteName = getTitleAndSiteName( state );
-
-	const titleParts = [];
+	let title = '';
 
+	const unreadCount = getUnreadCount( state );
 	if ( unreadCount ) {
-		titleParts.push( '(' + unreadCount + ')' );
-	}
-
-	if ( titleAndSiteName ) {
-		titleParts.push( titleAndSiteName );
+		title += `(${ unreadCount }) `;
 	}
 
-	const leftTitlePart = titleParts.join( ' ' );
-	return [ decodeEntities( leftTitlePart ), 'WordPress.com' ].join( ' — ' );
-}
-
-function getTitleAndSiteName( state ) {
-	const siteSpecificGroups = [ 'sites', 'editor' ];
-	const title = getTitle( state );
-	const siteId = getSelectedSiteId( state );
-	const siteTitle = getSiteTitle( state, siteId );
-	const titleParts = [];
-
-	if ( title ) {
-		titleParts.push( title );
-	}
-
-	// Display site name as title part only if we're in 'My Sites'
-	if ( includes( siteSpecificGroups, getSectionGroup( state ) ) && siteId ) {
-		titleParts.push( siteTitle );
-	}
+	title += [
+		getTitle( state ),
+		isSiteSection( state ) && getSiteTitle( state, getSelectedSiteId( state ) )
+	].filter( Boolean ).join( ' ‹ ' );
```

Comment:
Oh wow, didn't know that `.filter( Boolean )` trick. Love it. Much more concise now than with all the `if`s and `push`es.